Userland/Utilities/checksum.cpp

```diff
@@ -64,12 +65,8 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         while (!file->eof() && !file->has_error())
             hash.update(file->read(PAGE_SIZE));
         auto digest = hash.digest();
-        auto digest_data = digest.immutable_data();
-        StringBuilder builder;
-        for (size_t i = 0; i < hash.digest_size(); ++i)
-            builder.appendff("{:02x}", digest_data[i]);
-        auto hash_sum_hex = builder.build();
-        outln("{}  {}", hash_sum_hex, path);
+        auto bytes = ReadonlyBytes(digest.immutable_data(), digest.data_length());
+        outln("{}  {}", encode_hex(bytes), path);
```

Comment:
AK::Format supports this with the `:hex-dump` format specifier, perhaps consider using that?

Reply:
I have too look at it but even if it works it doesn't seems to be the right format option.
`hex-dump` looks more like a debug feature, instead I would use `x` or `X` to say it needs an hexadecimal representation of `ReadOnlyBytes` with respecting case...

Reply:
`hex-dump` just prints the hexadecimal representation of its argument, optionally with an actual hexdump, it's not really specific to debugging: https://serenity.godbolt.org/z/Kh164rTzW

Reply:
Thanks for the URL, it helped me to toy more dynamically with the library, and it seems I was miss-leaded while reading the code. I'm adapting the change set right now.
In a a future patch, I'll propose a change to add an enum to vectorize format casing handling, and address upper case ':hex-dump', since some formats *requires* them.